src/Aspire.Hosting.Azure/AzureResourcePreparer.cs

```diff
@@ -180,11 +180,21 @@ private async Task BuildRoleAssignmentAnnotations(DistributedApplicationModel ap
 
                         if (resource != identityResource)
                         {
-                            // attach the identity resource to compute resource so it can be used by the compute environment
-                            resource.Annotations.Add(new AppIdentityAnnotation(identityResource));
+                            // Only add the AppIdentityAnnotation if the resource doesn't already have one
```

Comment:
This code is only reached when `EnvironmentSupportsTargetedRoleAssignments` returns true - e.g. When calling AddAzureContainerAppEnvironment or AppService.

So if someone calls `WithAzureUserAssignedIdentity`, it isn't going to work and they won't get an indication as azd will manage the identity of the apps.

We should update the check to throw an exception if either it encounters role assignments, or AppIdentityAnnotations without EnvironmentSupportsTargetedRoleAssignments being true.  (Maybe update that method name too?)